server/src/test/java/org/candlepin/audit/EventBuilderTest.java

```diff
@@ -49,4 +55,14 @@ public void testSetEventDataForPoolSetsEventDataJson() {
         Event event = eventBuilder.setEventData(pool).buildEvent();
         assertEquals(expectedEventData, event.getEventData());
     }
+
+    @Test
+    public void testSetEventDataForNonModifiedEventsThrowsIesException() {
+        Pool pool = mock(Pool.class);
+        eventBuilder = new EventBuilder(factory, Event.Target.POOL, Event.Type.CREATED);
+
+        exceptions.expect(IseException.class);
+        exceptions.expectMessage("This method is only for type MODIFIED Events.");
```

Comment:
You can also accomplish this with the test annotation without setting the rule globally if you don't care about the exception details (which, generally speaking, we don't).

```
@Test(expected = IseException.class)
public void test...
```

Reply:
I'm aware of that option, but in this case I think we do care, because the same type of exception (IseException) can be thrown further down the call stack (since setEventData(entity, entity) calls setEventData(entity), which can throw IseException too). We don't want the test to pass for the wrong reason.

Reply:
That's fine. Just make sure you aren't testing implementation details. We don't want to have to rewrite our tests if the method's underlying implementation is changed.